src/ui/counterExamplesView.ts

```diff
@@ -12,111 +12,120 @@ interface IColorOptions {
+        ? await this.getCounterexamplesDebounced(params)
+        : await this.languageClient.getCounterexamples(params);
       if(!this.disposed) {
-        this.showCounterExamples(counterExamples, editor);
+        this.showCounterexamples(counterexamples, editor);
       }
     } catch(error: unknown) {
       if(!(error instanceof DebounceError)) {
-        window.showErrorMessage(`CounterExample request failed: ${error}`);
+        window.showErrorMessage(`Counterexample request failed: ${error}`);
       }
     }
   }
 
-  private hideCounterExamples(editor: TextEditor): void {
+  private hideCounterexamples(editor: TextEditor): void {
     const document = editor.document.uri;
     const decoration = this.activeDecorations.get(document);
     if(decoration == null) {
       return;
     }
     decoration.dispose();
     this.activeDecorations.delete(document);
+    this.synchronizeContextMenu(document);
+  }
+
+  private synchronizeContextMenu(document: Uri | boolean) {
+    const canShowCounterexamples = typeof document === 'boolean' ? document : !this.activeDecorations.has(document);
+    commands.executeCommand('setContext', 'dafny.counterexampleMenu.CanShowCounterExample', canShowCounterexamples);
   }
 
-  private showCounterExamples(counterExamples: ICounterExampleItem[], editor: TextEditor): void {
+  private showCounterexamples(counterexamples: ICounterexampleItem[], editor: TextEditor): void {
     const document = editor.document.uri;
     if(this.activeDecorations.has(document)) {
       return;
     }
-    const decorators = counterExamples
-      .filter(counterExample => counterExample.position.line >= 0)
-      .map(counterExample => CounterExamplesView.createDecorator(counterExample));
-    const decoration = CounterExamplesView.createTextEditorDecoration();
+    const decorators = counterexamples
+      .filter(counterexample => counterexample.position.line >= 0)
+      .map(counterexample => CounterexamplesView.createDecorator(counterexample));
+    const decoration = CounterexamplesView.createTextEditorDecoration();
     this.activeDecorations.set(document, decoration);
+    this.synchronizeContextMenu(document);
     editor.setDecorations(decoration, decorators);
   }
 
-  private static createDecorator(counterExample: ICounterExampleItem): DecorationOptions {
-    const contentText = Object.entries(counterExample.variables)
+  private static createDecorator(counterexample: ICounterexampleItem): DecorationOptions {
+    const contentText = Object.entries(counterexample.variables)
       .map(([ name, value ]) => `${name} = ${value}`)
       .join(' ');
     // TODO earlier versions showed a warning that there are references present.
-    const line = counterExample.position.line;
+    const line = counterexample.position.line;
     return {
       range: new Range(
-        new Position(line, counterExample.position.character + 1),
+        new Position(line, counterexample.position.character + 1),
```

Comment:
Do you know where the typechecking warnings come from?

Reply:
Yes. The newest version of VSCode considers positions to be `uinteger` instead of `number`. I added a cast.